proxy/ParentYamlConfig.h

```diff
@@ -0,0 +1,106 @@
+/** @file
+
+  Implementation of Parent Yaml Config parsing
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "ProxyConfig.h"
+#include "yaml-cpp/yaml.h"
+
+using namespace std;
```

Comment:
Could just be:
```
using std::string;
```

Reply:
We really don't want to do this in a header file. It makes it very challenging for any other code to include this.

Reply:
this has been re-implemented.